service/javadsl/client/src/main/scala/com/lightbend/lagom/internal/javadsl/client/JavadslWebSocketClient.scala

```diff
@@ -3,22 +3,31 @@
  */
 package com.lightbend.lagom.internal.javadsl.client
 
-import javax.inject.{ Inject, Singleton }
+import javax.inject.{ Inject, Singleton, Provider }
 
-import com.lightbend.lagom.internal.client.WebSocketClient
+import com.lightbend.lagom.internal.client.{ WebSocketClient, WebSocketClientConfig }
+import com.typesafe.config.Config
 import io.netty.channel.EventLoopGroup
 import play.api.Environment
 import play.api.inject.ApplicationLifecycle
 
 import scala.concurrent.ExecutionContext
 
 @Singleton
-class JavadslWebSocketClient(environment: Environment, eventLoop: EventLoopGroup,
-                             lifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext) extends WebSocketClient(environment, eventLoop, lifecycle) with JavadslServiceApiBridge {
+class WebSocketClientConfigProvider @Inject() (configuration: Config) extends Provider[WebSocketClientConfig] {
+
+  lazy val get = {
+    WebSocketClientConfig(configuration)
+  }
+}
+
+@Singleton
+class JavadslWebSocketClient(environment: Environment, config: WebSocketClientConfig, eventLoop: EventLoopGroup,
+                             lifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext) extends WebSocketClient(environment, config, eventLoop, lifecycle) with JavadslServiceApiBridge {
 
   // Constructor that manages its own event loop
   @Inject
-  def this(environment: Environment, applicationLifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext) = {
-    this(environment, WebSocketClient.createEventLoopGroup(applicationLifecycle), applicationLifecycle)
+  def this(environment: Environment, config: WebSocketClientConfig, applicationLifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext) = {
+    this(environment, config, WebSocketClient.createEventLoopGroup(applicationLifecycle), applicationLifecycle)
   }
```

Comment:
Maybe a simpler solution would be to make this overloaded constructor take a `Config` instead of a `WebSocketClientConfig` and call:

```
this(environment, WebSocketClientConfig(config), WebSocketClient.createEventLoopGroup(applicationLifecycle), applicationLifecycle)
```

Then the provider is not needed.